lib/records/P_RecCore.cc

```diff
@@ -507,6 +529,7 @@ RecReadStatsFile()
   ink_rwlock_wrlock(&g_records_rwlock);
 
   if ((m = RecMessageReadFromDisk(snap_fpath)) != nullptr) {
```

Comment:
Looks like it already reads before you do the check

Reply:
The problem occurs during writing(sync) but not reading which is called before writing. I put the check here if this file is found.

Reply:
Presumably reading is called _after_ writing was called in a previous process instance...

Reply:
I need to double check on this